pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -292,6 +293,12 @@ func constructDatastoreContext(config types.DownloaderConfig, status *types.Down
 			downloadURL = downloadURL + "/" + config.Name
 		}
 	}
+	if dst.CipherInfo != nil && len(dst.CipherPassword) != 0 {
+		password, err := cipherDecrypt(dst.CipherInfo, dst.CipherPassword)
+		if len(password) != 0 && err == nil {
+			dst.Password = password
```

Comment:
Ditto - use a local variable since you can't permanently modify the config. The config is owned and published by zedagent.